scripts/src/public/services/orders.ts

```diff
@@ -241,35 +243,34 @@ export async function getOrderHistory(
 	};
 }
 
-function orderDbToApi(order: db.ExternalOrder): ExternalOrder;
-function orderDbToApi(order: db.MarketplaceOrder): MarketplaceOrder;
-function orderDbToApi(order: db.ExternalOrder | db.MarketplaceOrder): ExternalOrder | MarketplaceOrder {
+function orderDbToApi(order: db.MarketplaceOrder | db.ExternalOrder): Order {
 	if (order.status === "opened") {
 		throw new Error("opened orders should not be returned");
 	}
 
 	const base = {
 		id: order.id,
 		offer_id: order.offerId,
+		offer_type: order.type,
 		status: order.status,
-		error: order.error,
-		completion_date: (order.currentStatusDate || order.createdDate).toISOString(), // XXX should we separate the dates?
+		amount: order.amount,
 		title: order.meta.title,
 		description: order.meta.description,
-		amount: order.amount
+		completion_date: (order.currentStatusDate || order.createdDate).toISOString(), // XXX should we separate the dates?
+		blockchain_data: order.blockchainData,
+		error: order.error,
 	};
 
 	return order.isMarketplaceOrder() ?
 		Object.assign(base, {
-			offer_type: order.type,
+			result: order.getValue(),
 			content: order.meta.content,
-			blockchain_data: order.blockchainData,
 			call_to_action: order.meta.call_to_action,
 		}) :
 		Object.assign(base, {
-			blockchain_data: {
-				recipient_address: order.blockchainData!.recipient_address
-			}
+			result: order.getValue(),
+			content_type: "confirm_payment",
+			call_to_action: "click to view in app",
```

Comment:
1. > get rid of MarketplaceOrder and ExternalOrder
2. I made a mistake about `content_type`. It's only defined for Offer but not defined for Order. What I meant was the result itself contains a `type` which is either `coupon` or `payment` (see my PR https://github.com/kinfoundation/ecosystem-api/pull/16/files#diff-fe030a7c1568b3decf599edf399be7f4R344)
3. see what I sent you on slack:
```typescript
function orderDbToApi(order: db.Order): Order {
    if (order.status === "opened") {
        throw new Error("opened orders should not be returned");
    }

    const base = {
        id: order.id,
        offer_id: order.offerId,
        offer_type: order.type,
        status: order.status,
        amount: order.amount
        title: order.meta.title,
        description: order.meta.description,
        call_to_action: order.meta.call_to_action,
        completion_date: (order.currentStatusDate || order.createdDate).toISOString(), // XXX should we separate the dates?
        content: order.meta.content,  // will be empty for external order
        blockchain_data: order.blockchainData,
        error: order.error,  // will be null for anything other than "failed"
        result: order.value,  // will be a coupon code or a confirm_payment JWT
    };
}
```